### PR TITLE
Adfgvx cipher implementation

### PR DESCRIPTION
**Pull request by avastmick**
Implementation of `ADFGVX` Cipher. Basic function is chain the existing `Polybius` and `ColumnarTransposition` ciphers.

Tests included:
- `encrypt_message`
- `decrypt_message`
- `with_utf8`
- `invalid_key_phrase`

Most tests are delegated to those in `Polybius` and `ColumnarTransposition` ciphers.

Updated `.travis.yml` to include `rustfmt` check to ensure consistent code style. Build fails if developer has not formatted code, use: `cargo +nightly fmt`

Ran `Clippy` - `cargo +nightly clippy` to lint existing code. I've fixed up most of the issues and it largely aligns with idiomatic styles.

Merged in latest `master` and merged in my working `adfgvx` branch.

Known issues:
- Code smell: `ColumnarTransposition` pads with whitespace that leads to output that can include trailing whitespace. I *think* the cipher would not include whitespace, but without the padding, the `decrypt` function is harder as it is iterating over ragged column lengths that if unhandled with cause out of bounds errors.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,21 +28,22 @@ The crypt only contains a few ciphers, but with time (and your help) it will hav
...
+- [x] Hill
+- [x] Fractionated Morse
+- [x] Columnar Transposition
+- [ ] Baconian
+- [ ] Porta
 - [ ] Homophonic
 - [ ] Four-Square
-- [x] Hill
 - [ ] Playfair
 - [ ] ADFGVX
```
Comment (arosspope) [inline README.md]: Mark as completed.

File: src/adfgvx.rs [hunk 2]
```diff
@@ -1,304 +1,182 @@
...
-    ///     ['A','B','G','D','E','F'])).unwrap();
+    /// use cipher_crypt::{Cipher, ADFGVX};
     ///
-    /// assert_eq!("BCdfdfbcbdgf 🗡️ dfgcbf bfbcbzdf ezbcacac",
-    ///    p.encrypt("Attack 🗡️ the east wall").unwrap());
+    /// let a = ADFGVX::new((
+    ///     "ph0qg64mea1yl2nofdxkr3cvs5zw7bj9uti8".to_string(),
+    ///     "GERMAN".to_string(),
+    /// )).unwrap();
+
```
Comment (arosspope) [inline src/adfgvx.rs]: Missing `///` ?

File: src/adfgvx.rs [hunk 3]
```diff
@@ -1,304 +1,182 @@
...
     ///
-    /// assert_eq!("Attack 🗡️ the east wall",
-    ///    p.decrypt("BCdfdfbcbdgf 🗡️ dfgcbf bfbcbzdf ezbcacac").unwrap());
+    /// let a = ADFGVX::new((
+    ///        "ph0qg64mea1yl2nofdxkr3cvs5zw7bj9uti8".to_string(),
+    ///        "GERMAN".to_string(),
+    ///    )).unwrap();
+    ///
+    /// let cipher_text =
+    ///             concat!(
```
Comment (arosspope) [inline src/adfgvx.rs]: Can you make sure the indenting is consistent with the encrypt documentation?

**Comment by avastmick** [comment]
Okay great. To be honest, this was my first Rust project, so apologies for the sloppiness in places. I'll fix up.

I'll have a look at the `ColumnarTransposition` code and the white space padding. In some tests I ran there was legitimate trailing whitespace. I understand why you included the padding, it makes the subsequent processing so much easier. But, unfortunately, I feel it does create downstream problems. I had a quick go a resolving, but I ran out of time (holidays here in China, and the kids are at home looking for things to do!).

Cheers

File: README.md [hunk 4]
```diff
@@ -26,21 +28,22 @@ The crypt only contains a few ciphers, but with time (and your help) it will hav
...
+- [x] Fractionated Morse
+- [x] Columnar Transposition
+- [ ] Baconian
+- [ ] Porta
 - [ ] Homophonic
 - [ ] Four-Square
-- [x] Hill
 - [ ] Playfair
-- [ ] ADFGVX
+- [x] ADFGVX
```
Comment (arosspope) [inline README.md]: Make sure to move this entry up so its underneath the other completed ciphers.

**Comment by avastmick** [comment]
Changes to README are in. Nice attention to detail! 👍

**Comment by arosspope** [comment]
Did you want to fix the trailing whitespace issue in another PR? We could always merge this one and create an issue for someone to fix later - up to you though.

**Comment by avastmick** [comment]
Please - another PR for the whitespace, I've run out of time for now and probably won't get back to it until next week. I'll sort that out then. Cheers!

I'll open an Issue.

**Pull request merged**